web-ui/src/api/api.js

```diff
@@ -58,6 +58,24 @@ export const getMyFetch = async () => {
   return myFetch;
 };
 
+function windowLogin() {
+  return new Promise((resolve, reject) => {
+    const authUrl = `${BASE_API_URL}/login?close=true`;
+    const loginWindow = window.open(authUrl, "Login", "width=500,height=600");
+
+    const interval = setInterval(() => {
+      try {
+        if (loginWindow.closed) {
+          clearInterval(interval);
+          resolve();
+        }
+      } catch (err) {
+        reject(err);
+      }
+    }, 1000);
+  });
+}
+
```

Comment:
I think this is only workable for localhost, isn't it?

Reply:
I think it works elsewhere because the URL doesn't really matter (except for our route match to close the window).  In production, it will see that the JWT has expired (or doesn't exist) and will make the user pick their google account to use.